### ISSUE TITLE
"Expected an Expression" with Empty Curly Brace Initializer 

### ISSUE THREAD

**Issue by Craie** [microsoft/vscode-cpptools#7872]
Bug type: Language Service

**Describe the bug**
- OS and Version:
   macOS 11.4

- VS Code Version:
   1.58.2

- C/C++ Extension Version:
   1.5.1

- Other extensions you installed (and if the issue persists after disabling them):
   none

- If using SSH remote, specify OS of remote machine:
   n/a

- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).
   When using a pair of empty curly braces to default construct/initialize a variable/object, I get the error "expected an expression" at the close curly.  This is a single file project.  Nothing complex.
   
   If I do Cmd-Shift-B to build and run it in the Terminal, all is fine.  So Intellisense seems to be marking this for itself.

   I've got every standard indicator set to C++14.
   
   I've got clang 12.0.5 installed (Mac/Xcode variant, of course).

**Steps to reproduce**
1. Code something like:  char u{};
2. Save.
3. See error squiggle.

**Expected behavior**
 - I expected no error squiggle or message.

**Code sample and logs**

- Code sample
   int main()
   {
       char u{};
   }

- Configurations in `c_cpp_properties.json`
{
    "configurations": [
        {
            "name": "Mac",
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "defines": [],
            "macFrameworkPath": [
                "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks"
            ],
            "compilerPath": "/usr/bin/clang++",
            "cStandard": "c17",
            "cppStandard": "c++14",
            "intelliSenseMode": "macos-clang-x64"
        }
    ],
    "version": 4
}

- Logs from running `C/C++: Log Diagnostics` from the VS Code command palette
   I don't know what this is or how to do it.

- Logs from [the language server logging](https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server)
   I don't know what this is or how to do it.

**Screenshots**
 - I hope it isn't necessary, but just ask if you need one.

**Additional context**
 - n/a

**Comment by Colengms** [comment]
Hi @Craie .  It does seem like this error is due to the file not being interpreted as `c++14`.  Could you run the `C/C++: Log Diagnostics` command from the VS Code command palette, with the repro file open, and provide the output?  That should indicate how the file was configured for IntelliSense.

**Comment by Craie** [comment]
I'm not sure what that means.  What's the command palette?  I looked all through the menus and couldn't find either this or the C/C++: Log Diagnostics.  I'm clearly missing something, but what?

**Comment by Colengms** [comment]
Hi @Craie .  The command palette can be opened via Command-Shift-P by default on Mac (Control-Shift-P on other platforms), or via the View menu -> Command Palette.  A great number of features of VS Code and extensions are accessible primarily via the command palette.

**Comment by Craie** [comment]
Aha!  Thank you.  And I found it!  I'd put an old set of command-line flags in two places.  These included a -ansi.  When I discovered it, I removed it from the set in tasks.json, but I didn't think about the other one since it didn't seem to be doing anything.  I'll take that out now.  Thank you for teaching me about the palette!  And I'm sorry to have wasted a bug number for this.  *shakes head*  *bows*
